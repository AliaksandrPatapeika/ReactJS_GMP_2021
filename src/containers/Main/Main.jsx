import './Main.less';

import React from 'react';

import ResultCount from '../../components/ResultCount';
import {MockResultCount} from '../../mocks/mockData';
import Menu from '../Menu';
import MoviesList from '../MoviesList';

const Main = () => (
  <main className="mainContainer">
    <Menu />
    <ResultCount count={MockResultCount} />
    <MoviesList />
  </main>
);

export default Main;
