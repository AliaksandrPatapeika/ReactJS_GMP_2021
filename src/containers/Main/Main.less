@import "../../less/variables";

.mainContainer {
	padding: 0 62px;
	background-color: @black;

	&.disabled {
		display: none;
	}

	.loading {
		padding: 40px 0;
		color: @red;
		font-size: 40px;
		font-family: @NetflixSans-Medium;
		text-align: center;
	}
}