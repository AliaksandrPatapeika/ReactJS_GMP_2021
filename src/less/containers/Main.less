@import "../variables";

.mainContainer {
    background-color: @black;
        padding: 0 62px;
}