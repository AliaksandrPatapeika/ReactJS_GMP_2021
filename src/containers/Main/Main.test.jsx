import React from 'react';
import renderer from 'react-test-renderer';

import Main from './Main';

describe('Main', () => {
  test('renders correctly', () => {
    const main = renderer
      .create(<Main />)
      .toJSON();

    expect(main).toMatchSnapshot();
  });
});
