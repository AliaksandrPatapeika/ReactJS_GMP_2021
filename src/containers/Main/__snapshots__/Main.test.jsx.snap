// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<main
  className="mainContainer"
>
  <div
    className="menuContainer"
  >
    <nav>
      <ul
        className="resultsFilterContainer"
      >
        <li
          className="filterItem active"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          ALL
        </li>
        <li
          className="filterItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          DOCUMENTARY
        </li>
        <li
          className="filterItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          COMEDY
        </li>
        <li
          className="filterItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          HORROR
        </li>
        <li
          className="filterItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          CRIME
        </li>
      </ul>
    </nav>
    <div
      className="resultsSortContainer"
    >
      <div
        className="sortBy"
      >
        SORT BY
      </div>
      <div
        className="releaseDate"
      >
        RELEASE DATE
      </div>
      <button
        className="arrow"
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="Sort Button"
          src="test-file-stub"
        />
      </button>
    </div>
  </div>
  <div
    className="resultCountContainer"
  >
    <span
      className="resultCount"
    >
      39
       
    </span>
    <span>
      movies found
    </span>
  </div>
  <div
    className="moviesListContainer"
  >
    <div
      className="movieCardContainer"
    >
      <img
        alt="Item"
        className="itemImage"
        src="test-file-stub"
      />
      <button
        className="threeDotsIcon"
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="MovieButton"
          src="test-file-stub"
        />
      </button>
      <div
        className="descriptionContainer"
      >
        <div
          className="itemTitle"
        >
          Pulp Fiction
        </div>
        <div
          className="itemReleaseDate"
        >
          2004
        </div>
      </div>
      <div
        className="itemGenre"
      >
        Action & Adventure
      </div>
    </div>
    <div
      className="movieCardContainer"
    >
      <img
        alt="Item"
        className="itemImage"
        src="test-file-stub"
      />
      <button
        className="threeDotsIcon"
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="MovieButton"
          src="test-file-stub"
        />
      </button>
      <div
        className="descriptionContainer"
      >
        <div
          className="itemTitle"
        >
          Bohemian Rhapsody
        </div>
        <div
          className="itemReleaseDate"
        >
          2003
        </div>
      </div>
      <div
        className="itemGenre"
      >
        Drama, Biography, Music
      </div>
    </div>
    <div
      className="movieCardContainer"
    >
      <img
        alt="Item"
        className="itemImage"
        src="test-file-stub"
      />
      <button
        className="threeDotsIcon"
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="MovieButton"
          src="test-file-stub"
        />
      </button>
      <div
        className="descriptionContainer"
      >
        <div
          className="itemTitle"
        >
          Kill Bill: Vol 2
        </div>
        <div
          className="itemReleaseDate"
        >
          1994
        </div>
      </div>
      <div
        className="itemGenre"
      >
        Oscar winning Movie
      </div>
    </div>
    <div
      className="movieCardContainer"
    >
      <img
        alt="Item"
        className="itemImage"
        src="test-file-stub"
      />
      <button
        className="threeDotsIcon"
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="MovieButton"
          src="test-file-stub"
        />
      </button>
      <div
        className="descriptionContainer"
      >
        <div
          className="itemTitle"
        >
          Avengers: War of Infinity
        </div>
        <div
          className="itemReleaseDate"
        >
          2004
        </div>
      </div>
      <div
        className="itemGenre"
      >
        Action & Adventure
      </div>
    </div>
    <div
      className="movieCardContainer"
    >
      <img
        alt="Item"
        className="itemImage"
        src="test-file-stub"
      />
      <button
        className="threeDotsIcon"
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="MovieButton"
          src="test-file-stub"
        />
      </button>
      <div
        className="descriptionContainer"
      >
        <div
          className="itemTitle"
        >
          Inception
        </div>
        <div
          className="itemReleaseDate"
        >
          2003
        </div>
      </div>
      <div
        className="itemGenre"
      >
        Action & Adventure
      </div>
    </div>
    <div
      className="movieCardContainer"
    >
      <img
        alt="Item"
        className="itemImage"
        src="test-file-stub"
      />
      <button
        className="threeDotsIcon"
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="MovieButton"
          src="test-file-stub"
        />
      </button>
      <div
        className="descriptionContainer"
      >
        <div
          className="itemTitle"
        >
          Reservoir dogs
        </div>
        <div
          className="itemReleaseDate"
        >
          1994
        </div>
      </div>
      <div
        className="itemGenre"
      >
        Oscar winning Movie
      </div>
    </div>
  </div>
</main>
`;
